jetty-websocket/websocket-jetty-client/src/main/java/module-info.java

```diff
@@ -20,11 +20,7 @@
 {
     exports org.eclipse.jetty.websocket.client;
 
-    requires org.eclipse.jetty.client;
-    requires org.eclipse.jetty.http;
-    requires org.eclipse.jetty.io;
-    requires org.eclipse.jetty.util;
-    requires org.eclipse.jetty.websocket.jetty.api;
+    requires transitive org.eclipse.jetty.client;
     requires org.eclipse.jetty.websocket.core;
-    requires org.eclipse.jetty.websocket.jetty.common;
+    requires transitive org.eclipse.jetty.websocket.jetty.common;
```

Comment:
Why `requires transitive` for common? I thought common was supposed to be an internal thing, and the `transitive` is for implied readability.

Reply:
There are `-common` classes in signatures of `-client` classes, so `-client` cannot work without `-common`: every application that requires `-client` would also need to require `-common`.
To avoid such double require, `-client` transitively requires `-common`.